config/constructor_types.go

```diff
@@ -29,29 +27,35 @@ var (
 	pubKeyType    = reflect.TypeOf((*crypto.PubKey)(nil)).Elem()
 	pstoreType    = reflect.TypeOf((*peerstore.Peerstore)(nil)).Elem()
 	connGaterType = reflect.TypeOf((*connmgr.ConnectionGater)(nil)).Elem()
+	upgraderType  = reflect.TypeOf((*transport.Upgrader)(nil)).Elem()
 
 	// concrete types
-	peerIDType   = reflect.TypeOf((peer.ID)(""))
-	upgraderType = reflect.TypeOf((*tptu.Upgrader)(nil))
-	pskType      = reflect.TypeOf((pnet.PSK)(nil))
+	peerIDType = reflect.TypeOf((peer.ID)(""))
+	pskType    = reflect.TypeOf((pnet.PSK)(nil))
 )
 
 var argTypes = map[reflect.Type]constructor{
-	upgraderType:  func(h host.Host, u *tptu.Upgrader, cg connmgr.ConnectionGater) interface{} { return u },
-	hostType:      func(h host.Host, u *tptu.Upgrader, cg connmgr.ConnectionGater) interface{} { return h },
-	networkType:   func(h host.Host, u *tptu.Upgrader, cg connmgr.ConnectionGater) interface{} { return h.Network() },
-	muxType:       func(h host.Host, u *tptu.Upgrader, cg connmgr.ConnectionGater) interface{} { return u.Muxer },
-	securityType:  func(h host.Host, u *tptu.Upgrader, cg connmgr.ConnectionGater) interface{} { return u.Secure },
```

Comment:
I removed the `muxType` and the `securityType`. All tests are passing, but I'm not sure this means that this is actually safe.

Reply:
i dont see why not.

Reply:
IIRC, the idea was that some transports may not need the entire upgrader, and may just want a multiplexer etc. But honestly, we might as well give them an upgrader.